hubblestack/extmods/fileserver/s3fs.py

```diff
@@ -575,6 +575,10 @@ def _read_buckets_cache_file(cache_file):
     with salt.utils.files.fopen(cache_file, 'rb') as fp_:
         try:
             data = pickle.load(fp_)
+            # check for 'corrupted' cache data ex: {u'base':[]}
+            if not list(filter(None, data.values())):
```

Comment:
Last one will work just fine I believe. First one won't since saltenv won't always be base and it doesn't cover the use case of having multiple saltenvs `{u'base': [], u'testing':[{u'somedict':[]}]}`

Reply:
I really hope people aren't using saltenv in their hubble ... it doesn't work very well in salt. It sure as heck won't work in hubble. I think the whole reference to 'base' is vestigial in this context. But whatever. As long as we're not using `filter()` and forcing our iterator down to a `list()`, we'll be fine.